pkg/driver/driver.go

```diff
@@ -72,3 +108,32 @@ func Register(logger *zap.SugaredLogger) {
 	// TODO(el): Don't discard but hide?
 	_ = mysql.SetLogger(log.New(ioutil.Discard, "", 0))
 }
+
+// ctxConnector adds driver.DriverContext support for drivers that do not implement it.
+type ctxConnector struct {
+	driver driver.Driver
+	name   string
+}
```

Comment:
Why is this needed? The MySQL driver seems to support `DriverContext`. At least the implementation of `Connect()` doesn't seem ideal as it doesn't stop any operations if the context is canceled but only checks when done. Thus, `context.WithTimeout(...)` won't really work as expected.

Reply:
The alternative would be to panic in `OpenConnector()` which is not an option for me as this code is library code. In the end, we're just wrapping here. I can't see how to change `Connect ()` any further as it's the driver that doesn't actually handle contexts.

Reply:
We could check the context before calling Open though.

Reply:
I see two more options:
1. Just always use the `DriverContext` interface here as `github.com/go-sql-driver/mysql` implements it. So `ctxConnector` shouldn't be used anyways at the moment and this could be asserted at compile time. However, `github.com/lib/pq` currently does not support that interface. There's a PR for that (https://github.com/lib/pq/pull/900) but it didn't make any progress since May 2020.
2. Execute `Open()` in a Goroutine so that connect will return immediately in case the context is canceled and if this happens, let the Goroutine close and throwaway the connection after `Open()` returns.

Given that this should be dead code right now anyways, I wouldn't bother with it now and see if we want to implement (2) when we add PostgreSQL support at a later point if that PR hasn't made any progress until then.